PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Purchases.swift
+//
+//  Created by Joshua Liebowitz on 8/18/21.
+
+// swiftlint:disable file_length function_parameter_count type_body_length
+import Foundation
+import StoreKit
+
+// MARK: Block definitions
+// NOTE: Any changes here must be reflected in PurchasesCoreSwift.h for ObjC compatibility.
+/**
+ Completion block for calls that send back a `PurchaserInfo`
+ */
+public typealias ReceivePurchaserInfoBlock = (PurchaserInfo?, Error?) -> Void
+
+/**
+ Completion block for `-[Purchases checkTrialOrIntroductoryPriceEligibility:completionBlock:]`
+ */
+public typealias ReceiveIntroEligibilityBlock = ([String: IntroEligibility]) -> Void
+
+/**
+ Completion block for `-[Purchases offeringsWithCompletionBlock:]`
+ */
+
+public typealias ReceiveOfferingsBlock = (Offerings?, Error?) -> Void
+
+/**
+ Completion block for `-[Purchases productsWithIdentifiers:completionBlock:]`
+ */
+public typealias ReceiveProductsBlock = ([SKProduct]) -> Void
+
+/**
+ Completion block for `-[Purchases purchaseProduct:withCompletionBlock:]`
+ */
+public typealias PurchaseCompletedBlock = (SKPaymentTransaction?, PurchaserInfo?, Error?, Bool) -> Void
+public typealias RCPurchaseCompletedBlock = PurchaseCompletedBlock
+
+/**
+ Deferred block for `purchases:shouldPurchasePromoProduct:defermentBlock:`
+ */
+public typealias DeferredPromotionalPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
+
+/**
+ * Deferred block for `-[Purchases paymentDiscountForProductDiscount:product:completion:]`
+ */
+@available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
+public typealias PaymentDiscountBlock = (SKPaymentDiscount?, Error?) -> Void
+
+/**
+ * `Purchases` is the entry point for Purchases.framework. It should be instantiated as soon as your app has a unique
+ * user id for your user. This can be when a user logs in if you have accounts or on launch if you can generate a random
+ * user identifier.
+ *  @warning Only one instance of Purchases should be instantiated at a time! Use a configure method to let the
+ *  framework handle the singleton instance for you.
+ */
+@objc(RCPurchases) public class Purchases: NSObject {
+
+    /**
+     * Delegate for `Purchases` instance. The delegate is responsible for handling promotional product purchases and
+     * changes to purchaser information.
+     */
+    @objc public var delegate: PurchasesDelegate? {
+        get { privateDelegate }
+        set {
+            privateDelegate = newValue
+            purchaserInfoManager.delegate = self
+            purchaserInfoManager.sendCachedPurchaserInfoIfAvailable(appUserID: appUserID)
+            Logger.debug(Strings.configure.delegate_set)
+        }
+    }
+
+    let subscriberAttributesManager: SubscriberAttributesManager
```

Comment:
private

Reply:
it should live along with the other injected dependencies, right? Perhaps we could add a `// mark`?

Reply:
We have one: `expect(purchases.subscriberAttributesManager).toNot(beNil())` in our tests so we must do `internal`

Reply:
Re: where it should live, I'm conflicted, since it must be `internal` I keep it with the other internals, but it probably makes more sense to be with the other injected deps....

Reply:
I think that test was there because we used to have a `RCPurchases+SubscriberAttributes` header file, that provided access to just the tests to that property...
but I don't think that particular test added much value, so I'd be happy to kill it. It'd provide consistency with the other properties, and in tests we typically access the `subscriberAttributesManager` instance that gets injected directly, instead of through the `purchases` object

Reply:
Word. Killing! 🔪 🩸